x/mongo/driver/topology/pool_test.go

```diff
@@ -23,6 +23,22 @@ func TestPool(t *testing.T) {
 	t.Run("newPool", func(t *testing.T) {
 		t.Parallel()
 
+		t.Run("minPoolSize should not exceed maxPoolSize", func(t *testing.T) {
+			t.Parallel()
+
+			p := newPool(poolConfig{MinPoolSize: 100, MaxPoolSize: 10})
+			assert.Equalf(t, uint64(10), p.minSize, "expected minSize of a pool to be smaller than maxSize")
+
+			p.close(context.Background())
+		})
+		t.Run("minPoolSize should not exceed maxPoolSize", func(t *testing.T) {
```

Comment:
I think you want a different name for this test. Something like `minPoolSize can exceed maxPoolSize of 0`? The assertion error message might have to change, too.